First stages of implementing the Digital agonizer.

MaxistheSpy opened the pull request:
Created Class for BlockDigitalAgonizer
Created Skeleton of ContainerDigitalAgonizer
Created digital_agonizer.json this is a temporary texture and should be revisited Added a logger to DeepMobLearningBM and a call to getModLog in the preInit stage (im not sure what this does) Refactored ModConstants to include ModInfo class for neatness Changes TileEntityDigitalAgonizer:
-updateCraftingState now always returns IDLE instead of null -> this is temporary to prevent NPE's until proper implementation of the method.

-Fully Implemented getContainer

Implemented full skeleton of DMLBM Registry inclduing: -Register blocks
-Register items: includes registering itemblocks
-Register Entities: currently does nothing
-Register Recipies: Im pretty sure this also does nothing as of now -Added loggers to each indicating when they start
-Registered DigitalAgonizer

Changes to DMLBMConfig include:
-Config for maximum RF capacity of agonizer currently defaults to MAXINT -Config for the agonizers maximum input RF/t also defaults to MAXINT Both of these were required for the Digital Agonizer constructor and i figured I might aswell make a config for it.

By the end of this the mob agonizer block exists and can be placed it also seems to not crash when interacted with. no gui or rightclick functionality yet.

MaxistheSpy committed:
First stages of implementing the Digital agonizer.
Created Class for BlockDigitalAgonizer
Created Skeleton of ContainerDigitalAgonizer
Created digital_agonizer.json this is a temporary texture and should be revisited
Added a logger to DeepMobLearningBM and a call to getModLog in the preInit stage (im not sure what this does)
Refactored ModConstants to include ModInfo class for neatness
Changes TileEntityDigitalAgonizer:
-updateCraftingState now always returns IDLE instead of null -> this is temporary to prevent NPE's until proper implementation of the method.

-Fully Implemented getContainer

Implemented full skeleton of DMLBM Registry inclduing:
-Register blocks
-Register items: includes registering itemblocks
-Register Entities: currently does nothing
-Register Recipies: Im pretty sure this also does nothing as of now
-Added loggers to each indicating when they start
-Registered DigitalAgonizer

Changes to DMLBMConfig include:
-Config for maximum RF capacity of agonizer currently defaults to MAXINT
-Config for the agonizers maximum input RF/t also defaults to MAXINT
Both of these were required for the Digital Agonizer constructor and i figured I might aswell make a config for it.

By the end of this the mob agonizer block exists and can be placed it also seems to not crash when interacted with. no gui or rightclick functionality yet.

MaxistheSpy committed:
First stages of implementing the Digital agonizer.
Created Class for BlockDigitalAgonizer
Created Skeleton of ContainerDigitalAgonizer
Created digital_agonizer.json this is a temporary texture and should be revisited
Added a logger to DeepMobLearningBM and a call to getModLog in the preInit stage (im not sure what this does)
Refactored ModConstants to include ModInfo class for neatness
Changes TileEntityDigitalAgonizer:
-updateCraftingState now always returns IDLE instead of null -> this is temporary to prevent NPE's until proper implementation of the method.

-Fully Implemented getContainer

Implemented full skeleton of DMLBM Registry inclduing:
-Register blocks
-Register items: includes registering itemblocks
-Register Entities: currently does nothing
-Register Recipies: Im pretty sure this also does nothing as of now
-Added loggers to each indicating when they start
-Registered DigitalAgonizer

Changes to DMLBMConfig include:
-Config for maximum RF capacity of agonizer currently defaults to MAXINT
-Config for the agonizers maximum input RF/t also defaults to MAXINT
Both of these were required for the Digital Agonizer constructor and i figured I might aswell make a config for it.

By the end of this the mob agonizer block exists and can be placed it also seems to not crash when interacted with. no gui or rightclick functionality yet.

MaxistheSpy committed:
First stages of implementing the Digital agonizer.
Created Class for BlockDigitalAgonizer
Created Skeleton of ContainerDigitalAgonizer
Created digital_agonizer.json this is a temporary texture and should be revisited
Added a logger to DeepMobLearningBM and a call to getModLog in the preInit stage (im not sure what this does)
Refactored ModConstants to include ModInfo class for neatness
Changes TileEntityDigitalAgonizer:
-updateCraftingState now always returns IDLE instead of null -> this is temporary to prevent NPE's until proper implementation of the method.

-Fully Implemented getContainer

Implemented full skeleton of DMLBM Registry inclduing:
-Register blocks
-Register items: includes registering itemblocks
-Register Entities: currently does nothing
-Register Recipies: Im pretty sure this also does nothing as of now
-Added loggers to each indicating when they start
-Registered DigitalAgonizer

Changes to DMLBMConfig include:
-Config for maximum RF capacity of agonizer currently defaults to MAXINT
-Config for the agonizers maximum input RF/t also defaults to MAXINT
Both of these were required for the Digital Agonizer constructor and i figured I might aswell make a config for it.

By the end of this the mob agonizer block exists and can be placed it also seems to not crash when interacted with. no gui or rightclick functionality yet.

bruberu commented on src/main/java/deepmoblearningbm/DMLBMConfig.java:
```diff
@@ -18,11 +18,19 @@ public static class MachineSettings {
+        @Config.Name("Digital Agonizer RF Capacity")
+        @Config.Comment("Energy capacity of Digital Agonizer in RF")
+        @Config.RangeInt(min = 0)
+        public int DIGITAL_AGONIZER_RF_CAPACITY = 2147483647;
```
This seems like an awfully high default RF capacity; perhaps make it something 80,000,000 RF?

MaxistheSpy replied:
800,000 now by default. I will adjust it once we implement RF costs.

bruberu commented on src/main/java/deepmoblearningbm/DMLBMConfig.java:
```diff
@@ -18,11 +18,19 @@ public static class MachineSettings {
+        @Config.Name("Digital Agonizer max input RF/t")
+        @Config.Comment("Maximum RF/t Digital Agonizer can accept")
+        @Config.RangeInt(min = 0)
+        public int DIGITAL_AGONIZER_RF_INPUTMAX = 2147483647;
```
This configuration doesn't seem to serve any purpose, other than to completely forbid certain models which need too much RF/t to work?

MaxistheSpy replied:
removed this config so its now just MAXINT always

bruberu commented on src/main/java/deepmoblearningbm/common/DMLBMRegistry.java:
```diff
@@ -1,13 +1,66 @@
+    //Items
+
+    //Blocks
+    public static final BlockDigitalAgonizer BLOCK_DIGITIAL_AGONIZER = new BlockDigitalAgonizer();
```
Digitial -> digital

MaxistheSpy replied:
Fixed

MaxistheSpy committed:
Merge remote-tracking branch 'origin/mx_digital_agonizer' into mx_digital_agonizer

# Conflicts:
#	src/main/resources/assets/deepmoblearningbm/models/block/digital_agonizer.json

MaxistheSpy committed:
WARNING: currently this commit causes crash upon looking at or placing digital agonizer I dont know why.

Uodated digital_agonizer_gui.png to new GUI

Created GuiDigitalAgonizer class

renamed various classes
ModConstants -> ModInfo
DMLBMRegistery -> DMARegsitry
DeepMobLearningBM -> DeepMobAgony
DMLBMConfig -> DMAConfig

Removed config for maximum input RF/t and set it to maxint
Set new default energy capacity of agonizer to
800,000rf to avoid infinite draw on network

MaxistheSpy committed:
WARNING: currently this commit causes crash upon looking at or placing digital agonizer I dont know why.

Uodated digital_agonizer_gui.png to new GUI

Created GuiDigitalAgonizer class

renamed various classes
ModConstants -> ModInfo
DMLBMRegistery -> DMARegsitry
DeepMobLearningBM -> DeepMobAgony
DMLBMConfig -> DMAConfig

Removed config for maximum input RF/t and set it to maxint
Set new default energy capacity of agonizer to
800,000rf to avoid infinite draw on network

MaxistheSpy committed:
Merge branch 'ThePansmith:main' into mx_digital_agonizer

MaxistheSpy committed:
Fixed crash in previous commit was due to DME dependency not being correctly updated.

Updated build.gradle to include a more up to date version of DME

Renamed the mods display name in ModInfo

MaxistheSpy committed:
Major commit.
Static part of gui should now be almost completely implemented

Broad strokes changes
Completely implemented the static parts of the gui as might be expected. Two slots in the top one for model and one for pristine matter.
Added the player inventory which will automatically center below the main gui.

changes in GuiDigitalAgonizer
  -Renamed TEXTURE to something more descriptive
  -Refactored HEIGHT -> TOTAL_HEIGHT
   and WIDTH -> TOTAL WIDTH to be more descriptive
   also changed their values
  -Implemented player inventory in the gui
  Very important that the formula be what it is so it stays centered. Im not sure why there isnt a version of drawPlayerInventory that does this automatically.
Added GuiTextures.java as of now im just adding it to force me to make a choice about weather to move all the gui textures to a single place or not.
Changes in TileEntityDigitalAgonizer.java
-added item handlers for things
-Pristine matter item handler should be temporary should change to something that handles any valid matter type for this block (probably).
-Copied over the capability adapter from LootFabricator, I dont know exactly how this works I do wonder if it isnt possible to make it more modular.
